homeassistant/components/sensor/speedtest.py

```diff
@@ -0,0 +1,105 @@
+"""
+homeassistant.components.sensor.speedtest
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Speedtest.net sensor based on speedtest-cli.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.speedtest/
+"""
+import logging
+import sys
+import re
+from datetime import timedelta
+from subprocess import check_output
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['speedtest-cli==0.3.4']
+_LOGGER = logging.getLogger(__name__)
+
+_SPEEDTEST_REGEX = re.compile(r'Ping:\s(\d+\.\d+)\sms\nDownload:\s(\d+\.\d+)'
+                              r'\sMbit/s\nUpload:\s(\d+\.\d+)\sMbit/s\n')
+
+SENSOR_TYPES = {
+    'ping': ['Ping', 'ms'],
+    'download': ['Download', 'Mbit/s'],
+    'upload': ['Upload', 'Mbit/s'],
+}
+
+# Return cached results if last scan was less then this time ago
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Setup the Speedtest sensor. """
+
+    data = SpeedtestData(hass.config.path)
+
+    dev = []
+    for variable in config['monitored_conditions']:
```

Comment:
The short answer is, because it is good form. Strings the user will have to put in should generally be defined in constants rather than hard coded. This is, generally speaking, because if they have to be changed later, they are up at the top and easier to locate. Additionally, if/when we tackle true localization, this string will change based on location. This isn't quite possible if it is hard coded.

Also, I would recommend making it `CONF_MONITORED_CONDITIONS` to maintain commonality with other components that do this correctly.